activejob/lib/active_job/queue_adapter.rb

```diff
@@ -61,6 +60,15 @@ def queue_adapter=(name_or_adapter)
       end
 
       private
+        def with_default_if_no_adapter_set
+          value = yield
+          if value.nil?
+            self.queue_adapter = :async
+            value = yield
+          end
+          value
+        end
```

Comment:
This method is really a bit weird. I think I'd rather see a much more straightforwrd:

```ruby
self.queue_adapter = :async if _queue_adapter.nil?
```

In both `queue_adapter` and `queue_adapter_name`.

Reply:
Yeah that was pretty funky. Thanks.